cypress/integration/dashboard/create_dashboard_spec.js

```diff
@@ -0,0 +1,36 @@
+describe('Create Dashboard', () => {
+  beforeEach(() => {
+    cy.login();
+    cy.visit('/dashboards');
+  });
+
+  it('create new', () => {
+    cy.getByTestId('CreateButton').click();
+    cy.get('li[role="menuitem"').contains('Dashboard').click();
+    cy.getByTestId('EditDashboardDialog').within(() => {
+      cy.getByTestId('DashboardSaveButton').should('be.disabled');
+      cy.get('input').type('Foo Bar');
+      cy.getByTestId('DashboardSaveButton').click();
+    });
+    cy.url().should('include', '/dashboard/foo-bar');
+
+    cy.visit('/dashboards');
+    cy.get('.table-main-title').contains('Foo Bar').should('exist');
+  });
+
+  it('archive', () => {
+    cy.get('.table-main-title[href*="dashboard/foo-bar"]').as('listItem');
+    cy.get('@listItem').click();
+    cy.getByTestId('DashboardMoreMenu')
+      .click()
+      .within(() => {
+        cy.get('li').contains('Archive').click();
+      });
+
+    cy.get('.btn-warning').contains('Archive').click();
+    cy.get('.label-tag-archived').should('exist');
+    
+    cy.visit('/dashboards');
+    cy.get('@listItem').should('not.exist');
```

Comment:
The `.as` seems to yield the same object as before ([ref](https://docs.cypress.io/api/commands/as.html#Yields)). If so, in this case is probably better to:
```javascript
const listItem = '.table-main-title[href*="dashboard/foo-bar"]';
cy.get(listItem).click();
// ...
cy.get(listItem).should('not.exist');
```

Reply:
If I understand correctly, `as` is similar to wrapping a sequence of Cypress commands in a function and when called, it runs the function against the current context.
Why do you think it's better to save the selector instead?

Reply:
I updated the test. Doesn't use this anymore. I do the assertion with snapshot instead.

Reply:
I initially thought this one would hold the same reference as before, therefore after a `cy.visit` this second `cy.get('@listItem')` would never exist. However they actually have this as [an example](https://docs.cypress.io/guides/core-concepts/introduction-to-cypress.html#Using-Aliases-to-Refer-to-Previous-Subjects). It seems it holds the same reference as before, but will run the function again if needed. Cypress has a lot of magic behind those commands :laughing:. With this in mind your approach is much better :rocket: